### PR TITLE
Remove notifications from HMI_API

### PR DESCRIPTION
**Pull request by indlu**
[HMI_API] OnPhoneCall, OnEmergencyEvent, OnDeactivateHMI removed

They were replaced by OnEventChanged notification

### CODE REVIEW

**Comment by indlu** [comment]
Please review @LevchenkoS, @dcherniev, @okozlovlux , @LuxoftAKutsan

**Comment by indlu** [comment]
Rebuild required

**Comment by indlu** [comment]
Rebuild required

**Comment by indlu** [comment]
Rebuild required

**Comment by indlu** [comment]
Please review @LevchenkoS, @dcherniev, @okozlovlux , @LuxoftAKutsan

File: src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc [hunk 1]
```diff
@@ -152,7 +152,8 @@ TEST_F(HMIMessageHandlerImplTest, RemoveHMIMessageAdapter_ExpectRemoved) {
 }
 
 // TODO(atimchenko) SDLOPEN-44 Wrong message to observer
-TEST_F(HMIMessageHandlerImplTest, OnMessageReceived_ValidObserver_Success) {
+TEST_F(HMIMessageHandlerImplTest,
```
Comment (LuxoftAKutsan) [inline src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc]: @indlu where is to do and jira issue?

**Pull request merged**